app/src/main/java/woowacourse/movie/view/widget/SeatTableLayout.kt

```diff
@@ -0,0 +1,111 @@
+package woowacourse.movie.view.widget
+
+import android.os.Bundle
+import android.widget.TableLayout
+import android.widget.TableRow
+import androidx.core.view.children
+import woowacourse.movie.R
+import woowacourse.movie.domain.seat.MovieSeatRow
+import woowacourse.movie.domain.seat.Seat
+import woowacourse.movie.domain.seat.Seats
+import woowacourse.movie.view.data.SeatTable
+import woowacourse.movie.view.data.SeatsViewData
+import woowacourse.movie.view.data.TableSize
+import woowacourse.movie.view.getSerializable
+import woowacourse.movie.view.mapper.MovieSeatMapper.toView
+
+class SeatTableLayout(
+    private val tableLayout: TableLayout,
+    override val saveStateKey: String
+) : SaveState {
+    var onSelectSeat: (SeatsViewData) -> Unit = {}
+    var seatSelectCondition: (Int) -> Boolean = { true }
+
+    fun selectedSeats(): SeatsViewData {
+        return tableLayout.children.flatMap { tableRow ->
+            ((tableRow as TableRow).children as Sequence<SeatView>).filter { seatView ->
```

Comment:
```suggestion
            ((tableRow as TableRow).children as Sequence<SeatView>).filter { seatView ->
```
`as Sequence<SeatView>` 부분에서 warning이 발생하고 있습니다.
warning이 왜 발생했는지 학습해보시고, warning이 발생하지 않도록 개선해보면 어떨까요? (이하 관련 내용 동일)

Reply:
일반적으로 상속관계에 있는 클래스에 대한 자동 캐스팅은 언어 단에서 지원되나, Collection의 내부 클래스에 대해서는 지원되지 않습니다. 이는 코틀린의 runtime type erasure 특성에 의한 것으로 런타임에 제네릭 클래스의 인스턴스에 타입 인자 정보가 들어있지 않습니다.

이를 해결하기 위해 `inline reified` 등의 키워드를 사용하는 방법이 있지만, scope function 내에서 타입을 캐스팅 하는 방법으로 해결하였습니다.